packages/multichain/src/scope/errors.ts

```diff
@@ -0,0 +1,48 @@
+import { JsonRpcError } from '@metamask/rpc-errors';
+
+/**
+ * Thrown when chains requested in a CAIP-25 `wallet_createSession` call are not supported by the wallet.
+ * Defined in [CAIP-25 error codes section](https://github.com/ChainAgnostic/CAIPs/blob/main/CAIPs/caip-25.md#trusted-failure-codes).
+ */
+export const REQUESTED_CHAINS_NOT_SUPPORTED_ERROR = new JsonRpcError(
```

Comment:
Hmm, this error would have a stack trace that points to this line. Not great if this error is used for debugging purposes, as we wouldn't know where it came from.

We solve this in `rpc-errors` by defining a builder function for each type of error, rather than exposing them as constants. Then the error's stack trace would point at the specific location where it was created.

Alternatively, we could just define constants for the error numbers and messages.

Reply:
I've switched these over to builder functions: https://github.com/MetaMask/core/pull/4784/commits/1abd454dcb02b26a9118c9560f09525ddf83e98c